pkg/operator/operator.go

```diff
@@ -99,12 +115,17 @@ func (c OpenShiftControllerManagerOperator) sync() error {
 		// we equally do not allow removed/unmanaged
 	case operatorapiv1.Unmanaged:
 		reasonString = fmt.Sprintf("%sUnsupported", string(operatorConfig.Spec.ManagementState))
-		messageString = fmt.Sprintf("the controller manager spec was set to %s state, but that is unsupported, and has no effect on this condition", string(operatorConfig.Spec.ManagementState))
-
-		// as we are ignoring / not supporting unmanaged/removed, we still process any other inputs to the sync/reconciliation
-		// unlike what we used to do
+		messageString = fmt.Sprintf(
+			"the controller manager spec was set to %s state, but that is unsupported, "+
+				"and has no effect on this condition",
+			string(operatorConfig.Spec.ManagementState),
+		)
+
+		// as we are ignoring / not supporting unmanaged/removed, we still process any other inputs
+		// to the sync/reconciliation unlike what we used to do
 	case operatorapiv1.Managed:
-		// we want to empty out the reason/message string if transitioning from the other phases so the default setting above is good
+		// we want to empty out the reason/message string if transitioning from the other phases so
+		// the default setting above is good
```

Comment:
Yeah in relations to my comment in the assertOperatorConditions method, we should validate that the messages field is emptied out when going back to managed

Reply:
For this specific case, detecting message fields have been emptied out, we will need to recur to inspect which actions have been registered in the (fake) client. We can teak the approach of testing to assert this test-case as well.

In the current implementation we only validate the final state of the resources, and `sync()` updates the cluster config status twice. The status update does not register a event, but we can still obtain this information in the client.